app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+# Use React, Bootstrap, CRA, and GQL for the Nimbus Front-End Stack
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
+
+* Rename `static/` to `legacy-ui/` since `static/rapid/` will become obsolete and `static/core/` will be the legacy Normandy platform
+* Move `templates/` into `legacy-ui/`
+* Use `create-react-app` to create a new React application in `app/experimenter/nimbus-ui/`
+
+
+### Front-End Stack
+
+* Continue using React, but:
+  * Use the `create-react-app` toolchain
+  * Use Typescript over Proptypes for more strict typing
+  * Use GraphQL for API calls and `apollo-client-cache` for global state management
```

Comment:
I might be missing context, but "Use GraphQL for API calls" seems to go beyond front-end stack concerns for this ADR. 

Given that [the Decision Outcome](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md#decision-outcome) over in ADR 0001 was to build on top of Experimenter, these GQL APIs would need be changes or new APIs in the Django app. Could also be supported by a node/TS GQL proxy server, but that's probably unwarranted complexity.

We have FxA-team experience with GQL in node/TS - have we looked into the same under Python & Django?

Reply:
Might be worth clarifying in the ADR – but we did agree to go with [graphene](https://github.com/graphql-python/graphene-django) for the back end. I think Jared and Ben are working on filing some of that this week

Reply:
This is a great call-out as I didn't realize the first ADR doesn't even mention GQL. I agree we should expound upon this a bit here.

@bbangert, I know you'd mentioned looking into [graphene-django](https://github.com/graphql-python/graphene-django). Does it seem like it's a relatively low level of effort to implement?

Reply:
> Might be worth clarifying in the ADR – but we did agree to go with graphene for the back end.

Yeah, I don't see mentions of graphene in this ADR or 0001, so that sounds like an Architectural Decision that wasn't Recorded. :) Otherwise, I'm scratching my head reading this as trying to further leverage FxA experience with GQL in node/TS that's not exactly applicable here.

Reply:
I think this has been resolved in updates to the ADR